src/main/java/org/kiwiproject/jersey/client/ServiceIdentifier.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    public ServiceIdentifier(String serviceName, String preferredVersion, Connector connector) {
+        this.serviceName = serviceName;
+        this.preferredVersion = preferredVersion;
+        this.connector = connector;
+    }
+
+    public ServiceIdentifier(String serviceName, Duration connectTimeout, Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    public ServiceIdentifier(String serviceName, Connector connector, Duration connectTimeout, Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.connector = connector;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    public ServiceIdentifier(String serviceName,
+                             String preferredVersion,
+                             Connector connector,
+                             Duration connectTimeout,
+                             Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.preferredVersion = preferredVersion;
+        this.connector = connector;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    public ServiceIdentifier(String serviceName,
+                             String preferredVersion,
+                             String minimumVersion,
+                             Connector connector,
+                             Duration connectTimeout,
+                             Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.preferredVersion = preferredVersion;
+        this.minimumVersion = minimumVersion;
+        this.connector = connector;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    /**
+     * Make a copy of {@code original).
+     */
+    public static ServiceIdentifier copy0f(ServiceIdentifier original) {
+        checkArgumentNotNull(original);
+
+        var copy = new ServiceIdentifier();
+        copy.serviceName = original.getServiceName();
+        copy.preferredVersion = original.getPreferredVersion();
+        copy.minimumVersion = original.getMinimumVersion();
+        copy.connector = original.getConnector();
+        copy.connectTimeout = original.getConnectTimeout();
+        copy.readTimeout = original.getReadTimeout();
+        return copy;
+    }
+
+    /**
+     * Return a boolean indicating whether this {@link ServiceIdentifier) prefers the latest available version
+     * of the service it describes. A non-existent (null) {@code version} indicates latest version.
+     */
+    public boolean isPreferLatestVersion() {
+        return isNull(preferredVersion);
+    }
+
+    /**
+     * Creates a copy of this instance but with the specified {@code serviceName}.
+     *
+     * @implNote Intentionally not using Lombok's {@code Wither) because it seems to require an all arguments
```

Comment:
Is this still true? Also with our newer version of Lombok it is the With annotation not Wither anymore.

Reply:
It may not be true if we switch to builder